software/webapp/src/main/java/org/apache/brooklyn/entity/proxy/AbstractControllerImpl.java

```diff
@@ -317,6 +331,8 @@ protected void computePortsAndUrls() {
 
         sensors().set(PROTOCOL, inferProtocol());
         sensors().set(MAIN_URI, URI.create(inferUrl()));
+        sensors().set(MAIN_URI_MAPPED_SUBNET, URI.create("http://" + sensors().get(SUBNET_ADDRESS) + ":" + sensors().get(PROXY_HTTP_PORT) + "/"));
+        sensors().set(MAIN_URI_MAPPED_PUBLIC, URI.create(inferUrlPublic()));
```

Comment:
@neykov a very good point, but for `AbstractController` it is unfortunately a special case. It wants to use the `domain` sensor for the hostname in the URL, if set. This was added for `GeoscalingDnsService` (and is also useful for things like AWS ELB).

We'd still like to advertise `MAIN_URI_MAPPED_PUBLIC`, but not with the "normal" logic. The same goes I think for `MAIN_URI_MAPPED_SUBNET`: the machine's subnet-address doesn't make sense if it's an AWS ELB that is being used. We want `MAIN_URI_MAPPED_SUBNET` to always be populated, so that other blueprints can use this without worrying about the type of load balancer.

This is probably worth further discussion. But in the mean time I suggest we merge this PR, so we get it working (important for the tutorials/examples that show a simple web-cluster being deployed to GCE and/or Azure).

I'll also look separately at improving `OnPublicNetworkEnricher` so that it can read the public IP from a sensor (rather than only working if the public mapping is registered in `PortForwardManager`, as is done for DNAT).